sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -165,7 +173,7 @@ case class InSubquery(values: Seq[Expression], query: ListQuery)
     if (values.length != query.childOutputs.length) {
       TypeCheckResult.TypeCheckFailure(
         s"""
-           |The number of columns in the left hand side of an IN subquery does not match the
+           |The number of columns in the left hand side of a predicate subquery does not match the
```

Comment:
IMHO, we should try and mention the exact type of expression ? So we should say "IN" if its a In subquery and ANY of its an any subquery ? I am thinking, predicate subquery may not be very obvious to the end user ?